src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -139,34 +34,82 @@ def read_slr_data_from_files(slr_data_dir: pathlib.Path) -> gpd.GeoDataFrame:
-    for file_path in slr_data_dir.glob("*.csv"):
-        # Read the CSV file into a pandas DataFrame using pyarrow
-        slr_region = csv.read_csv(file_path).to_pandas()
-        # Extract the region name from the file name and add it as a new column in the DataFrame
-        file_name = file_path.stem
-        start_index = file_name.find('projections_') + len('projections_')
-        end_index = file_name.find('_region')
-        region_name = file_name[start_index:end_index]
-        slr_region['region'] = region_name
-        # Append the DataFrame to the list
-        slr_nz_list.append(slr_region)
-        # Log that the file has been successfully loaded
-        log.info(f"Successfully loaded the '{file_path.name}' data file.")
-    # Concatenate all the dataframes in the list and add geometry column
-    slr_nz = pd.concat(slr_nz_list, axis=0).reset_index(drop=True)
-    geometry = gpd.points_from_xy(slr_nz['lon'], slr_nz['lat'], crs=4326)
-    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz, geometry=geometry)
-    # Convert all column names to lowercase
-    slr_nz_with_geom.columns = slr_nz_with_geom.columns.str.lower()
+    # Get the url where sea level rise files need downloading
+    url = 'https://zenodo.org/records/11398538/export/json'
+
+    # Create dictionary to store the sea level rise dataset
+    slr_nz_dict = {}
+    # Request export information json from Zenodo
+    response = requests.get(url)
+    response.raise_for_status()
+    export_json = response.json()
+    # Get necessary links from json file
+    for file_name, file_info in export_json['files']['entries'].items():
+        # Get file name without extension
+        file_name_without_extension = os.path.splitext(file_name)[0]
+        # Collect sea level rise dataset and storing into dictionary
+        slr_nz_dict[file_name_without_extension] = pd.read_csv(file_info["links"]["content"])
+
+        # Check response for errors
+        response.raise_for_status()
+
+    # Column name used to merge
+    left_column_name = 'Site ID'
+    right_column_name = 'site'
+    # Merge Site Details dataframe and Sea level projections tables WITH VLM
+    slr_nz_merge_vlm = pd.merge(
+        slr_nz_dict["NZ_VLM_final_May24"], slr_nz_dict["NZSeaRise_proj_vlm"],
+        left_on=left_column_name, right_on=right_column_name,
+        how='left', validate='1:m',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_vlm['add_vlm'] = True
+
+    # Merge Site Details dataframe and Sea level projections tables WITHOUT VLM
+    slr_nz_merge_novlm = pd.merge(
+        slr_nz_dict["NZ_VLM_final_May24"], slr_nz_dict["NZSeaRise_proj_novlm"],
+        left_on=left_column_name, right_on=right_column_name,
+        how='left', validate='1:m',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_novlm['add_vlm'] = False
+
+    # Concatenate all the dataframes (both WITH VLM and WITHOUT VLM)
+    slr_nz = pd.concat([slr_nz_merge_vlm, slr_nz_merge_novlm], axis=0).reset_index(drop=True)
```

Comment:
```suggestion
    slr_nz_df = pd.concat([slr_nz_merge_vlm, slr_nz_merge_novlm], axis=0).reset_index(drop=True)
```
It looks like this df isn't special compared to any of the below ones so we can keep reprocessing this one to reduce memory usage.